adobe_vipm/flows/sync/agreement.py

```diff
@@ -61,31 +56,44 @@ def __init__(
         agreement: dict,
         customer: dict,
         adobe_subscriptions: list[dict],
+        *,
+        dry_run: bool,
     ):
         self._mpt_client = mpt_client
         self._adobe_client = adobe_client
         self._agreement = agreement
         self._customer = customer
         self._adobe_subscriptions = adobe_subscriptions
+        self._dry_run = dry_run
         self._authorization_id: str = agreement["authorization"]["id"]
         self._seller_id: str = agreement["seller"]["id"]
         self._licensee_id: str = agreement["licensee"]["id"]
         self._adobe_customer_id: str = get_adobe_customer_id(self._agreement)
 
-    def sync(self, *, dry_run: bool, sync_prices: bool) -> None:  # noqa: C901
+    @property
+    def agreement_id(self) -> str:
+        """Return agreement id."""
+        return self._agreement["id"]
+
+    @property
+    def product_id(self) -> str:
+        """Return agreement product id."""
+        return self._agreement["product"]["id"]
+
+    def sync(self, *, sync_prices: bool) -> None:  # noqa: C901
```

Comment:
<img width="1816" height="1248" alt="Image" src="https://github.com/user-attachments/assets/1d079837-0f4b-44be-83c0-e1c1b58ef108" />
<img width="1724" height="1286" alt="Image" src="https://github.com/user-attachments/assets/1a25cfb5-7506-4ef5-b199-3c838df02602" />

@jentyk  are you sure?